server/v2/cometbft/abci.go

```diff
@@ -238,44 +253,117 @@ func (c *consensus[T]) maybeRunGRPCQuery(ctx context.Context, req *abci.QueryReq
 		handlerFullName = string(md.Input().FullName())
 	}
 
-	// special case for simulation as it is an external gRPC registered on the grpc server component
+	// special case for non-module services as they are external gRPC registered on the grpc server component
```

Comment:
let's extra all those cases in a seperate function maybe `func (c *consensus) handleExtraServices` or smth. Let's move this as well in `grpc.go`, so it is all at the same place.